pkg/operator/controller/controller_dns_configmap.go

```diff
@@ -106,14 +108,32 @@ func desiredDNSConfigMap(dns *operatorv1.DNS, clusterDomain string) (*corev1.Con
 		clusterDomain = "cluster.local"
 	}
 
+	upstreamResolvers := operatorv1.UpstreamResolvers{
+		Upstreams: []string{"/etc/resolv.conf"},
+		Policy:    operatorv1.SequentialForwardingPolicy,
+	}
+
+	if dns.Spec.UpstreamResolvers.Upstreams != nil && len(dns.Spec.UpstreamResolvers.Upstreams) > 0 {
+		upstreamResolvers.Upstreams = dns.Spec.UpstreamResolvers.Upstreams
+		if len(dns.Spec.UpstreamResolvers.Upstreams) == 1 && dns.Spec.UpstreamResolvers.Upstreams[0] == "" {
+			upstreamResolvers.Upstreams[0] = "/etc/resolv.conf"
+		}
+	}
```

Comment:
You don't need to check for nil before calling `len()` on an slice type. If it's `nil` then it will [return 0](https://pkg.go.dev/builtin#len).
```suggestion
	if len(dns.Spec.UpstreamResolvers.Upstreams) > 0 {
		upstreamResolvers.Upstreams = dns.Spec.UpstreamResolvers.Upstreams
		// if user sets the upstream resolver list to an empty string default to the host resolve config file.
		if len(dns.Spec.UpstreamResolvers.Upstreams) == 1 && dns.Spec.UpstreamResolvers.Upstreams[0] == "" {
			upstreamResolvers.Upstreams[0] = "/etc/resolv.conf"
		}
	}
```